src/NServiceBus.Metrics.ServiceControl.Tests/QueueLength/When_sending_message.cs

```diff
@@ -0,0 +1,199 @@
+            public ConcurrentQueue<IDictionary<string, string>> Headers2 { get; } = new ConcurrentQueue<IDictionary<string, string>>();
+            public MetricsContext Data { get; set; }
+        }
+
+        [Test]
+        public void Should_enhance_it_with_queue_length_properties()
+        {
+            var context = Scenario.Define<Context>()
+                .WithEndpoint<Receiver1>()
+                .WithEndpoint<Receiver2>()
+                .WithEndpoint<Sender>(c =>
+                {
+                    c.When(s =>
+                    {
+                        var a1 = Address.Parse(ReceiverAddress1);
+                        s.Send(a1, new TestMessage());
+                        s.Send(a1, new TestMessage());
+
+                        var a2 = Address.Parse(ReceiverAddress2);
+                        s.Send(a2, new TestMessage());
+                        s.Send(a2, new TestMessage());
+                    });
+                })
+                .WithEndpoint<MonitoringSpy>()
+                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2 && c.Data != null)
+                .Run();
+            AssertSequencesReported(context);
+        }
+
+        static void AssertSequencesReported(Context context)
+        {
+            var sessionIds = new[]
+            {
+                AssertHeaders(context.Headers1),
+                AssertHeaders(context.Headers2)
+            };
+
+            var data = context.Data;
+            var counters = data.Counters;
+            var counterTokens = counters.Where(c => c.Name.StartsWith("Sent sequence for"));
+
+            foreach (var counter in counterTokens)
+            {
+                var tags = counter.Tags;
+                var counterBasedKey = tags.GetTagValue("key");
+                var type = tags.GetTagValue("type");
+
+                CollectionAssert.Contains(sessionIds, counterBasedKey);
+                Assert.AreEqual(2, counter.Count);
+                Assert.AreEqual("queue-length.sent", type);
+            }
+        }
+
+        static string AssertHeaders(IProducerConsumerCollection<IDictionary<string, string>> oneReceiverHeaders)
+        {
+            const string keyHeader = "NServiceBus.Metrics.QueueLength.Key";
+            const string valueHeader = "NServiceBus.Metrics.QueueLength.Value";
+
+            var headers = oneReceiverHeaders.ToArray();
+
+            var sessionKey1 = headers[0][keyHeader];
+            var sessionKey2 = headers[1][keyHeader];
+            Assert.AreEqual(sessionKey1, sessionKey2, "expected sessionKey1 == sessionKey2");
+
+            var sequence1 = long.Parse(headers[0][valueHeader]);
+            var sequence2 = long.Parse(headers[1][valueHeader]);
+            Assert.AreNotEqual(sequence2, sequence1, "expected sequence1 != sequence2");
+            Assert.IsTrue(sequence1 == 1 || sequence2 == 1, "sequence1 == 1 || sequence2 == 1");
```

Comment:
Would

> CollectionAssert.AreEquivalent(new []{1,2}, new []{sequence1, sequence2});

be more readable than

>  Assert.AreNotEqual(sequence2, sequence1, "expected sequence1 != sequence2");
>   Assert.IsTrue(sequence1 == 1 || sequence2 == 1, "sequence1 == 1 || sequence2 == 1");

?

Reply:
Personally, I see no change. Both require stopping for a sec.

Reply:
I like `AreEquivalent` much more.

Reply:
I don't have a strong opinion here.